core/src/main/java/lucee/runtime/type/QueryImpl.java

```diff
@@ -1110,18 +1111,23 @@ public int getCurrentrow(int pid) {
 	 * 
 	 * @return string list
 	 */
+
 	public String getColumnlist(boolean upperCase) {
+		return getColumnlist(upperCase, ",");
+	}
+
+	public String getColumnlist(boolean upperCase, String delim) {
 		StringBuilder sb = new StringBuilder();
 		for (int i = 0; i < columnNames.length; i++) {
-			if (i > 0) sb.append(',');
+			if (i > 0) sb.append(delim);
 			sb.append(upperCase ? columnNames[i].getUpperString() : columnNames[i].getString());
```

Comment:
It would be much more performant to UPPERCASE the result once at the end, no?

Reply:
yeah, nice catch :)